packages/core/addon/components/navi-visualizations/table.js

```diff
@@ -66,63 +66,96 @@ export default Component.extend({
+   * @param {String} metricName
+   * @param {Object} totalRow
+   * @param {Object} column
+   * @param {String} type - `grandTotal` || `subtotal`
+   * @returns {*} sum
+   */
+  computeColumnTotal(data, metricName /*, totalRow, column, type*/) {
+    return data.reduce((sum, row) => {
+      let number = Number(row[metricName]);
+      if (!Number.isNaN(number)) {
+        return sum + number;
+      }
+      return sum;
+    }, 0);
+  },
+
   /**
    * @method _computeTotal
    * Compute total for all metrics in the data
    *
    * @private
    * @param {Array} data
-   * @param {String} type
+   * @param {String} type - `grandTotal` || `subtotal`
    * @returns {Object} totalRow
    */
   _computeTotal(data, type) {
-    let columns = get(this, 'columns');
+    let columns = get(this, 'columns'),
+      hasPartialData = get(this, 'totalRows') > get(this, 'rowsInResponse');
 
-    return columns.reduce((totRow, column) => {
+    let totalRow = columns.reduce((totRow, column) => {
       //if dateTime set type
       if (column.type === 'dateTime') {
         set(totRow, column.field.dateTime, HEADER_TITLE[type]);
       }
 
       //set subtotal dimension if subtotal row
-      if (column.field.dimension === get(this, 'selectedSubtotal') && type === 'subtotal') {
+      if (
+        column.field.dimension === get(this, 'selectedSubtotal') &&
+        type === 'subtotal'
+      ) {
         let idField = `${column.field.dimension}|id`,
-            descField = `${column.field.dimension}|desc`;
+          descField = `${column.field.dimension}|desc`;
 
         set(totRow, idField, data[0][idField]);
         set(totRow, descField, data[0][descField]);
       }
 
-      //add meta to indicate totalRow
-      set(totRow, '__meta__', { isTotalRow: true });
-
-      //if partial data do not compute metric totals
-      if (get(this, 'totalRows') > get(this, 'rowsInResponse')) {
-        set(totRow, '__meta__.hasPartialData', true);
-        return totRow;
-      }
-
-      //if metric find sum
-      if (column.type === 'metric') {
+      //if metric and not partial data compute totals
+      if (column.type === 'metric' && !hasPartialData) {
         let metricName = canonicalizeMetric(column.field);
-        totRow[metricName] = data.reduce((sum, row) => {
-          let number = Number(row[metricName]);
-          if (!Number.isNaN(number)) {
-            return sum + number;
-          }
-          return sum;
-        }, 0);
+        totRow[metricName] = this.computeColumnTotal(
+          data,
+          metricName,
+          totRow,
+          column,
+          type
+        );
       }
 
       return totRow;
     }, {});
+
+    //add totalRow indication to meta
+    set(
+      totalRow,
+      '__meta__',
+      Object.assign({}, get(totalRow, '__meta__'), { isTotalRow: true })
+    );
+
+    //if partial data add indication to meta
+    if (hasPartialData) {
+      set(totalRow, '__meta__.hasPartialData', true);
```

Comment:
What are we doing with this metadata? Do we still want to display the totals even if we have partial data?

Reply:
Currently we're not displaying totals if we have partial data, I haven't changed this behavior. Only moved it out of the reduce function, setting it once per row is enough.